gtda/tests/test_common.py

```diff
@@ -0,0 +1,89 @@
+# License: GNU AGPLv3
+from functools import partial
+
+import pytest
+from sklearn.utils.estimator_checks import parametrize_with_checks
+
+from gtda.images.preprocessing import Binarizer, Inverter
+
+
+# mark checks to skip
+SKIP_TESTS = {
+  'Binarizer':  [],
+  'Inverter':  [],
+}
+
+# mark tests as a known failure
+# TODO: these should be addressed later.
+XFAIL_TESTS = {
+  'Binarizer':  ["check_transformer_data_not_an_array",
+                 "check_transformer_general",
+                 "check_transformer_general(readonly_memmap=True)",
+                 "check_fit2d_predict1d",
+                 "check_fit1d"],
+  'Inverter':  ["check_transformer_data_not_an_array",
+                "check_transformer_general",
+                "check_transformer_general(readonly_memmap=True)",
+                "check_fit2d_predict1d",
+                "check_fit1d"],
+}
+
+# adapted from sklearn.utils.estimator_check v0.22
+def _get_callable_name(obj):
```

Comment:
Basically this and the above function are necessary because we need a reliable way to convert both `check` and `estimator` objects to strings, and this is the way it's done in scikit-learn [here](https://github.com/scikit-learn/scikit-learn/blob/93776b9f529b7c26950e517deae4aa3ccdd9fe1a/sklearn/utils/estimator_checks.py#L312). I realize they are a bit complex.

Reply:
Great, thanks! Makes sense to me.